Add polling monitor flag for blazar Floating ip and new dry run flag for polling monitor

AnishReddyRavula opened the pull request:
- This is to enable the floating ip polling to clean up the resources in an errored or deleted lease
- * Setting the flag `enable_polling_monitor` to False then blazar does not periodically invoke the resource (FIP) monitor plugin to heal/clean the failed resources - PR for [FIP clean up](https://github.com/ChameleonCloud/blazar/pull/93)
- * When this option is not set, the monitor fails with following error - `oslo_config.cfg.NoSuchOptError: no such option enable_notification_monitor in group [virtual:floatingip]`

- Add config opt to blazar monitor to enable dry run
-  * Setting the flag `dry_polling_monitor` to True is to warn a failed/needs-fixing resource without attempting to fix it, to make sure that the fix is reliable and the monitor is actually picking up the right resource that needs fixing and the right fix is chosen

AnishReddyRavula committed:
Add plling monitor flag for blazar Floating ip

This is to enable the floating ip polling to clean up the resources in a errored or deleted lease

codyhammock reviewed (approved):
👍🏻

msherman64 commented on kolla/node_custom_config/blazar.conf:
```diff
@@ -58,6 +58,7 @@ enable_polling_monitor = true
 
 [virtual:floatingip]
 billrate = {{ blazar_floatingip_billrate }}
+enable_polling_monitor = true
```
What is the default behavior if this is not set?
Also, can you link to the blazar PR that depends on this flag?

AnishReddyRavula replied:
If `enable_polling_monitor` is False for FIPs, then it won't use the FIP monitor plugin to heal resources ( in this case FIPs)

msherman64 replied:
Right, but what if it's not set at all?
Just want to be sure that blazar will start successfully if an older config is in use.

AnishReddyRavula replied:
The monitor is set to run periodically and whenever it tries to run monitor, it fails with the following error
```oslo_config.cfg.NoSuchOptError: no such option enable_notification_monitor in group [virtual:floatingip]```

AnishReddyRavula committed:
Add config opt to blazar monitor to enable dry run

This is to warn a failed/needs-fixing resource without attempting to fix it, to make sure that fix is realiable and the monitor is actually picking up the right resource that needs fixing and the right fix is choosen

msherman64 commented on kolla/node_custom_config/blazar.conf:
```diff
@@ -47,6 +47,7 @@ heartbeat_in_pthread = false
 before_end = email
 email_relay = {{ blazar_email_relay }}
 enable_polling_monitor = true
+dry_polling_monitor = true
```
This means that dry-run is enabled by default?
(optional) Maybe we can change the name of the flag to `enable_polling_monitor_dry_run`?

I would request that we set this via a variable, so that it can be set from defaults.yml, otherwise a site will need to use their own node_custom_config to override.

Something like:
```
enable_polling_monitor = {{ blazar_physical_polling_monitor }}
dry_polling_monitor = {{ blazar_physical_polling_monitor_dry_run }}
```

msherman64 commented on kolla/node_custom_config/blazar.conf:
```diff
@@ -58,6 +59,8 @@ enable_polling_monitor = true
 
 [virtual:floatingip]
 billrate = {{ blazar_floatingip_billrate }}
+enable_polling_monitor = true
```
We should add a variable for this also, e.g. 
```
enable_polling_monitor = {{ blazar_fip_polling_monitor }}
dry_polling_monitor = {{ blazar_fip_polling_monitor_dry_run }}
```

AnishReddyRavula committed:
Set polling monitor via variable in blazar.conf and Rename dry run polling monitor

Set polling monitor and dry run enable/disable from a variable from defaults.yaml instead of using their own node_constom_config

msherman64 commented on kolla/defaults.yml:
```diff
@@ -76,6 +76,10 @@ blazar_network_default_resource_properties: '["=", "\$stitch_provider", "none"]'
+blazar_physical_polling_monitor: true
+blazar_physical_polling_monitor_dry_run: true
+blazar_fip_polling_monitor: true
+blazar_fip_polling_monitor_dry_run: true
```
just add `blazar_zun_polling_monitor: true` and I think this is good.

AnishReddyRavula committed:
Also add zun polling monitor in defaults.yml
